File: MethodTrackerTool.Tests/MethodLoggerTests.cs
using MethodTracker.MockProject;
using MethodTrackerTool;

namespace MethodTracker.Tests;

public class MethodLoggerTests
{
    [Fact]
    public async Task Sample() =>
        await MethodLogger.InitializeAsync("Sample", async () =>
            await new OrderService().ProcessOrderAsync(new OrderRequest
            {
                UserId = 13,
                ProductIds = [1, 4, 55, 342, 33, 334, 864, 268, 1042],
                TotalAmount = 20
            }));
}

File: MethodTrackerTool.Tests.VisualBasic/MethodLoggerTests.vb

Imports MethodTrackerTool.VisualBasic.MockProject
Imports Xunit

Public Class MethodLoggerTests

    <Fact>
    Public Async Function Sample() As Task
        MethodLogger.Initialize()

        Await New OrderService().ProcessOrderAsync(New OrderRequest With {
                                                      .UserId = 13,
                                                      .ProductIds = New List(Of Integer)({1, 4, 55, 342, 33, 334, 864, 268, 1042}),
                                                      .TotalAmount = 20
                                                      })

        MethodLogger.PrintJson()
    End Function

End Class

